### DISCUSSION TITLE
Requirements for Client Listings

### DISCUSSION THREAD

**Discussion by thornbill** [jellyfin/jellyfin-meta#27] [General]
The point of what should and should not be included on the client listing page on jellyfin.org and/or the docs site came up recently and we really don't have clear guidelines currently. This attempts to gather our discussion points into a clear list of rules for listing clients that we can then debate and eventually solidify in the docs.

### Requirements for listing as a recommended app

* The application must be a first-party application (meaning published and maintained by the Jellyfin team with source freely licensed and available in the Jellyfin GitHub organization).
**OR**
* The application must fill a significant void in the current first-party application offerings. Must be a high-quality application on a popular platform.

### Requirements for listing in all apps

* Any third-party application meeting the following requirements:
  * Must be aligned with the [Jellyfin Community Standards](https://jellyfin.org/docs/general/community-standards.html)
    * In particular the application must **NOT** engage in, encourage, or facilitate piracy.
  * Must adhere to the [Jellyfin Branding Guidelines](https://jellyfin.org/docs/general/contributing/branding.html) including usage of the Jellyfin name.
    * This includes usage of the Jellyfin name or `org.jellyfin` namespace that could hinder the ability to publish an official application to a store in the future.
  * Must include first rate support for Jellyfin servers. (i.e. Support for Jellyfin should be a primary function or at the same level of integration of any other supported services.)
  * Must **NOT** be specific to or intended to promote a specific hosted Jellyfin server instance.
  * Must have clear licensing and be void of any known issues related to attribution, copyright, or license violations.

To help with discovery, we should also include a link to the docs page on the main jellyfin.org client page.

**Comment by joshuaboniface** [comment]
One thing I think happens is that there's a perceived disparity between the "Main Site" + "Documentation" lists that might cause some friction. I'd like to propose two potential options, and I prefer the first:

1. All clients on the main site page, no separate docs page. Clear headers for the "Official Apps" (maintained by the Jellyfin team), perhaps with bigger icons, and for "3rd Party Apps" (everything else) with smaller icons. The latter list would of course be open to just about anything. 

2. We keep the split of "3rd part in docs", but clearly link this after the main list of official clients in its own header.

With both options though, I'd like to declutter the Clients page. Currently the icons are all *huge*, and take up a lot of screen real estate. I like the grid layout but 3 wide seems way too big to me. Maybe 6-8 wide would be nicer and then we could fit a lot more in with limited vertical space, and make the others more obvious. And the current headings are OK but also very large. Basically what I'm getting at is that we already have to scroll through at least 3 screen heights to get down the list of official apps; I'd like to reduce that to 1.

**Reply by Shadowghost** [reply]
I do think the 1st approach is fine if we enforce some restrictions on what 3rd party clients are in this case.
I'm thinking mainly about
1. The application is either a new project or a fork of an abandoned 1st party project
2. Any forks of one of our main applications for a specific platform should instead try to upstream the required changes - if that isn't possible from a technical point of view and the application basically consists of a patchset ontop of official code, proper attribution and especially proper versioning is a must and we should actively check and enforce that before.

I'm mentioning the 2nd point here because in this specific case we need to be especially careful on enforcing the branding and versioning guidelines to prevent issues from that fork to come into the main repo.

**Reply by thornbill** [reply]
We could also just call the sections featured (or recommended/suggested) and additional clients and stick to the rules I listed for each section. It probably does make more sense for them to all be listed in a single location since it should be less to maintain and easier for people to discover.

Maybe "additional clients" could be listed in a table to be efficient with space/layout.

**Reply by iwalton3** [reply]
I agree the clients page could use some re-work to make it easier to browse. I suspect there is still somewhat of a visibility issue since clients regularly need to be suggested in chats/online.

Adding third party clients to the list as a separate list with a disclaimer or as a link to “additional clients” in the docs both seem fine. I think that applications that are still in initial development and don’t work belong on the docs page only, ideally in a separate listing.

**Reply by iwalton3** [reply]
> 2\. Any forks of one of our main applications

I would be especially concerned about branded copies that seem to be “specific to” or “intended to drive traffic to” specific paid server instances. I have seen a few of those pop up, sometimes without even changing the maintainer names.

Oh and of course we want to avoid linking to anything that could be malicious or that is spyware.

**Reply by h1dden-da3m0n** [reply]
Something of more a concept idea than actual guideline proposal, as I agree with all concepts proposed so far:

I recently dove into the world of SBOM (due to work) and stumbled over cycloneDX and their [tool cenrter](https://cyclonedx.org/tool-center/).
From a general concept point of view I feel like integrating something similar to this for 3rd party clients could help users find a client if they don't find it in the first party clients.

i.e. If a user is looking for an open-source 3rd party client for platform X they could filter a list (assuming we provide 3rd party clients as an interactive list)

**Reply by mcarlton00** [reply]
I think the biggest issues with putting everything on the home page are:

* If we want it to look nice, then we have to have somewhat consistent logos.  And a lot of our 3rd party clients just straight up don't have logos at all
* Duplication of information.  We've learned over and over again that if we don't have one source of truth, they inevitably become out of sync and will end up with both places having totally independent information.

Not saying this to be opposed to it, but a warning that it will involve more effort to maintain and have it look reasonable.  That being said, the page already looks weird because some things have multiple links and some have one and the spacing is all fucky around those with more links.

**Comment by thornbill** [comment]
I have updated the rules in the original post based on feedback. I think the best approach is to have a single clients page on jellyfin.org with two sections (recommended and all apps). The docs site can just link to the main site to avoid duplicate effort. The recommended apps could use cards similar to the current page and all apps could be represented in a table that is filterable/sortable based on different criteria (platform, license, etc.).

**Comment by BotBlake** [comment]
Hey nice.
I think this is actually solved.
We have a pretty solid new Listing guideline in the making: https://github.com/jellyfin/jellyfin.org/pull/1833
And we do separate between Official/ 3rd Party.
